anchor/network/src/handshake/mod.rs

```diff
@@ -31,226 +26,148 @@ pub enum Error {
-    },
+pub struct Completed {
+    pub peer_id: PeerId,
+    pub their_info: NodeInfo,
 }
 
-/// Network behaviour handling the handshake protocol.
-pub struct Behaviour {
-    /// Request-response behaviour for the handshake protocol.
-    pub(crate) behaviour: RequestResponseBehaviour<Codec>,
-    /// Keypair for signing envelopes.
-    keypair: Keypair,
-    /// Local node's information provider.
-    node_info_manager: NodeInfoManager,
-    /// Events to emit.
-    events: Vec<Event>,
+#[derive(Debug)]
+pub struct Failed {
+    pub peer_id: PeerId,
+    pub error: Box<Error>,
 }
 
-impl Behaviour {
-    pub fn new(keypair: Keypair, local_node_info: NodeInfoManager) -> Self {
-        // NodeInfoProtocol is the protocol.ID used for handshake
-        const NODE_INFO_PROTOCOL: &str = "/ssv/info/0.0.1";
-
-        let protocol = StreamProtocol::new(NODE_INFO_PROTOCOL);
-        let behaviour =
-            RequestResponseBehaviour::new([(protocol, ProtocolSupport::Full)], Config::default());
-
-        Self {
-            behaviour,
-            keypair,
-            node_info_manager: local_node_info,
-            events: Vec::new(),
-        }
-    }
-
-    /// Create a signed envelope containing local node info.
-    fn sealed_node_record(&self) -> Envelope {
-        let node_info = self.node_info_manager.get_node_info();
-        node_info.seal(&self.keypair).unwrap()
-    }
-
-    fn verify_node_info(&mut self, node_info: &NodeInfo) -> Result<(), Error> {
-        let ours = self.node_info_manager.get_node_info().network_id;
-        if node_info.network_id != *ours {
-            return Err(Error::NetworkMismatch {
-                ours,
-                theirs: node_info.network_id.clone(),
-            });
-        }
-        Ok(())
-    }
-
-    fn handle_handshake_request(
-        &mut self,
-        peer_id: PeerId,
-        request: Envelope,
-        channel: ResponseChannel<Envelope>,
-    ) {
-        // Handle incoming request: send response then verify
-        let response = self.sealed_node_record();
-        let _ = self.behaviour.send_response(channel, response.clone()); // Any error here is handled by the InboundFailure handler
-
-        self.unmarshall_and_verify(peer_id, &request);
-    }
-
-    fn handle_handshake_response(&mut self, peer_id: PeerId, response: &Envelope) {
-        self.unmarshall_and_verify(peer_id, response);
-    }
-
-    fn unmarshall_and_verify(&mut self, peer_id: PeerId, envelope: &Envelope) {
-        let mut their_info = NodeInfo::default();
+pub fn create_behaviour(keypair: Keypair) -> Behaviour {
+    // NodeInfoProtocol is the protocol.ID used for handshake
+    const NODE_INFO_PROTOCOL: &str = "/ssv/info/0.0.1";
 
-        if let Err(e) = their_info.unmarshal(&envelope.payload) {
-            self.events.push(Event::Failed {
-                peer_id,
-                error: Error::NodeInfo(e),
-            });
-        }
+    let protocol = StreamProtocol::new(NODE_INFO_PROTOCOL);
```

Comment:
nit: `NODE_INFO_PROTOCOL` seems to just be used here so I'd  just do:
```suggestion
    let protocol = StreamProtocol::new("/ssv/info/0.0.1");
```